### DISCUSSION TITLE
Cofiguration for new rlink 2 rlan16

### DISCUSSION THREAD

**Discussion by sdade86** [cedricp/ddt4all#1060] [Q&A]
Hello all gentlemen!
I decided to mount a used r-link2 system, version lanr16 281152288R hw: 3.11.5.4 sw: 7.8.24.136 instead of the original lanr14 on my Talisman Initiale 2016.
Once installed I immediately noticed that I was quite lucky as it turned on without any problems with the anti-theft device.
But now there are problems. the joypad in the central console doesn't work at all while of the steering wheel controls only the button to lower the volume and change the source works.
I have already done several tests with ddt4all but I can't get it to work correctly.
Furthermore, I cannot find the correct xml file as the program always tells me that the control unit software does not match perfectly.
Please let me know if you need any other information and how to get it.
Thank you all!

**Comment by sdade86** [comment]
Just au update, joypad workin after can clip parameter update

**Comment by sdade86** [comment]
another proble is the hu display not showed in rlink2 setting

**Comment by shrlnm** [comment]
@sdade86 https://www.drive2.ru/b/643019689764615344/

**Comment by sdade86** [comment]
> @sdade86 https://www.drive2.ru/b/643019689764615344/

@shrlnm 

Thanks a lot for the link! It is very helpfull! Put the ecu in Virgin mode is mandatory for use macro? Or tha macro can config anyway? Other question is that: the radnav actually have different vin from car. I need to change it for macro? (If yes maybe i have answer for first question).
does the macro also restore the original audio parameters?

**Comment by sdade86** [comment]
Hello, today i have try the bot but not give me any file. Bot sayd data is ready and stop

**Comment by shrlnm** [comment]
@sdade86 You need to generate macro for your vin with your current 76: option. 
Virgin mode require in some cases when handsfree not working for example. 
Try to start without virgin. If something will not work then try with virgin mode.

I just checked the bot and it's working. Try once again. If you will get no success then share your vin

**Comment by sdade86** [comment]
@shrlnm Thanks for your hep. I have try again now but nothing. 
this is the command /vin VFR1RFD00854631851 76:281152288R
for vin and ecu ref
![Immagine](https://github.com/cedricp/ddt4all/assets/141132439/67764962-2e64-4109-83a4-48cb703fdaa8)

**Comment by sdade86** [comment]
I just realized I made a mistake when I wrote the wine... sorry, I'm really stupid!

**Comment by sdade86** [comment]
Ok, i have run macro created from the bot and hud come back! thanks a lot! but swc have same problem, you think need to put in virgin mode? 
Anf if need to put in virgin mode the best way is from developer menu? Or configuration reset from ddt4all?
I'm asking you because if I don't match the xml file perfectly I wouldn't want to do something wrong

**Comment by sdade86** [comment]
I'm back! yesterday I took a lot of courage and pressed the virgin button in dev mode. then I reconfigured with the bot macro and everything was perfect again (even the language in the notifications which before virgin mode did not match that choice)... everything except SWC. At this point I became convinced that it had to be a HW problem. So I re-unmounted rlink2 from the machine. I opened it and found the damage fortunately repairable with some soldering skills and some smd capacitors!

@shrlnm I really appreciate your work and your help, I would like to contribute with a donation, do you have paypal?  A thousand thanks

**Reply by shrlnm** [reply]
@sdade86 I am glad that my work helps. That's enough for me. I don't need money, and if you have a desire to thank me, then spend it on charity.

**Reply by Gsp29** [reply]
Buna, ma poti ajuta si pe mine cu o problema ?

**Reply by Gsp29** [reply]
> M-am întors! Ieri mi-am făcut mult curaj și am apăsat butonul virgin în modul dev. Apoi am reconfigurat cu macro-ul botului și totul a fost din nou perfect (chiar și limbajul din notificări, care înainte de modul virgin nu se potrivea cu acea alegere)... totul, cu excepția SWC. În acest moment m-am convins că trebuie să fie o problemă hardware. Așa că am demontat din nou rlink2 de pe mașină. L-am deschis și am constatat că, din fericire, deteriorarea putea fi reparată cu niște abilități de lipire și niște condensatoare smd!
> 
> @shrlnmApreciez foarte mult munca și ajutorul dumneavoastră, aș dori să contribui cu o donație, aveți PayPal? Mulțumesc mult.

ma poti ajuta si pe mine cu o problema

**Comment by sdade86** [comment]
Thanks to all for the contribute!
